platform/ios/src/MGLAnnotationImage.m

```diff
@@ -57,15 +57,11 @@ - (BOOL)isEqual:(id)other {
     
     return ((!_reuseIdentifier && !otherAnnotationImage.reuseIdentifier) || [_reuseIdentifier isEqualToString:otherAnnotationImage.reuseIdentifier])
     && _enabled == otherAnnotationImage.enabled
-    && ((!_image && !otherAnnotationImage.image) || [UIImagePNGRepresentation(_image) isEqualToData:UIImagePNGRepresentation(otherAnnotationImage.image)]);
```

Comment:
I see that `testAnnotationImage()` is failing as a result of my change. It’s true that drawing the image to a new context could make the two images unequal, but there are a number of reasons that would be the case anyways (such as different color profiles). In general, I don’t think a class’s `-isEqual:` implementation should be more thorough than the `-isEqual:` implementations of its ivars.